modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ipfinder/vm/TcpDiscoveryVmIpFinder.java

```diff
@@ -235,15 +246,28 @@ else if (ipStr.indexOf(':') >= 0)
 
                     return res;
                 }
-                catch (IllegalArgumentException | UnknownHostException e) {
+                catch (IllegalArgumentException e) {
                     throw new IgniteSpiException(errMsg, e);
                 }
             }
             else {
                 try {
                     int port = Integer.parseInt(portStr);
 
-                    return Collections.singleton(new InetSocketAddress(addrStr, port));
+                    Collection<InetSocketAddress> col = new LinkedHashSet<>();
+
+                    try {
+                        InetAddress[] inetAddresses = InetAddress.getAllByName(addrStr);
+
+                        for (InetAddress addrs : inetAddresses)
+                            col.add(new InetSocketAddress(addrs, port));
+                    }
+                    catch (UnknownHostException ignored) {
+                        //save previous behavior on UnknownHostException
+                        col = Collections.singleton(new InetSocketAddress(addrStr, port));
+                    }
+
+                    return col;
```

Comment:
What about multicast finder, btw? It has the same addresses input.

Reply:
The current fix is about VmIpFinder. Probably we should fix multicast too but ut should be done in a separate ticket.